overlord/devicestate/devicemgr.go

```diff
@@ -711,6 +711,18 @@ func (m *DeviceManager) ensureSeeded() error {
 	var opts *populateStateFromSeedOptions
 	if m.preseed {
 		opts = &populateStateFromSeedOptions{Preseed: true}
+		if !release.OnClassic {
+			// XXX: needed by loadDeviceSeed(), otherwise Ensure fails with:
+			// "devicemgr: cannot seed: internal error: requested inconsistent device seed: 20220105 (was )"
+			unloadDeviceSeed(m.state)
```

Comment:
I think the correct fix is to change preloadGadget to also take preseeding into account

Reply:
Thanks, fixed.